shopinvader_product_stock/models/product_product.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Akretion (http://www.akretion.com)
+# Copyright 2018 ACSONE SA/NV
+# Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo.addons.queue_job.job import job
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    @api.multi
+    @job(default_channel='root.search_engine.recompute_stock')
+    def _product_stock_update_all(self):
+        """
+        The goal of this function is to compute the new stock information
+        and update them in the data field. If data have change and the binding
+        is in done state we force it to 'to_update'
+        :return:
+        """
+        for record in self:
+            for binding in record.shopinvader_bind_ids:
+                if binding.sync_state == 'new':
+                    # this binding have been not yet computed
+                    # so we do not care to update it as it's not yet
+                    # on the site. The right stock qty will be exported
+                    # at it's first export
+                    continue
+                # I do not recommend to rename the stock key, but if you have
+                # a good reason to do it, do not worry we will use this key
+                # here
+                stock_key = binding._get_stock_key()
+                if not stock_key:
+                    continue
+                data = binding.data
+                if data[stock_key] != binding.stock_data:
+                    data[stock_key] = binding.stock_data
+                    vals = {'data': data}
+                    if binding.sync_state == 'done':
+                        vals['sync_state'] = 'to_update'
+                    binding.write(vals)
```

Comment:
@sebastienbeau With your changes we lost the `in real time`sync of stock level.  It's a very important feature.

Reply:
After thinking twice, I think you are right, depending of the customer case this feature maybe required. I think about adding an option to active or not the "real time stock".
If activated we directly export the binding here, if not we flag it 'to_update'.